docs/how-to/package-with-hatch.md

```diff
@@ -0,0 +1,385 @@
+
+```bash
+calculator
+```
+
+## Building 
+
+Building produces archive files that contain your code.
+When you install a package via pip or other tool, it will download one of these archives.
+
+To build your project with Hatch, change to the directory containing your `pyproject.toml` and run the `hatch build` subcommand:
+
+```
+cd textual-calculator
+hatch build
+```
+
+After a moment, you should find that Hatch has created a `dist` (distribution) folder, which contains the project archive files.
+You don't typically need to use these files directly, but feel free to have a look at the directory contents.
+
+!!! note "Packaging TCSS and other files"
+
+    Hatch will typically include all the files needed by your project, i.e. the `.py` files.
+    It will also include any Textual CSS (`.tcss`) files in the project directory.
+    Not all build tools will include files other than `.py`; if you are using another build tool, you may have to consult the documentation for how to add the Textual CSS files.
+
+
+## Publishing
+
+After your project has been successfully built you are ready to publish it to PyPI.
+
+If you don't have a PyPI account, you can [create one now](https://pypi.org/account/register/).
+Be sure to follow the instructions to validate your email and set up 2FA (Two Factor Authentication).
+
+Once you have an account, login to PyPI and go to the Account Settings tab.
+Scroll down and click the "Add API token" button.
+In the "Create API Token" form, create a token with name "Uploads" and select the "Entire project" scope, then click the "Create token" button.
+
+Copy this API token (long string of random looking characters) somewhere safe.
+This API token is how PyPI authenticates uploads are for your account, so you should never share your API token or upload it to the internet.
+
+Run the following command (replacing `<YOUR API TOKEN>` with the text generated in the previous step):
+
+```bash
+hatch publish -u __token__ -a <YOUR API TOKEN>
+```
+
+Hatch will upload the distribution files, and you should see a PyPI URL in the terminal.
+
+### Managing API Tokens
+
+Creating an API token with the "all projects" permission is required for the first upload.
+You may want to generate a new API token with permissions to upload a single project when you upload a new version of your app (and delete the old one).
+This way if your token is leaked, it will only impact the one project.
+
+### Publishing new versions
+
+If you have made changes to your app, and you want to publish the updates, you will need to update the `version` value in the `__about__.py` file, then repeat the build and publish steps.
+
+!!! tip "Managing version numbers"
+
+    See [Semver](https://semver.org/) for a popular versioning system (used by Textual itself).
+
+## Installing the calculator
+
+From the user's point of view, they only need run the following command to install the calculator:
+
+```bash
+pip install textual_calculator
```

Comment:
I understand that this is here mostly to later go on and explain why it's a bad idea, but it does feel like it's advertising this approach to some extent.

If I'm following along as a beginner, I'm probably going to run this command because you haven't told me it's a terrible idea yet. Maybe we shouldn't be sharing the actual snippet here 🤷